File: go-leetcode/linked-list/234.回文链表.go
package leetcode

func isPalindrome(head *ListNode) bool {
	// 1. 将链表值存入切片
	values := []int{}
	curr := head
	for curr != nil {
		values = append(values, curr.Val)
		curr = curr.Next
	}

	// 2. 双指针检查回文
	for i, j := 0, len(values)-1; i < j; i, j = i+1, j-1 {
		if values[i] != values[j] {
			return false
		}
	}
	return true
}

File: python-leetcode/linked-list/234.回文链表.py
from typing import Optional

from type import ListNode


class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        # 1. 将链表值存入列表
        values: list[int] = []
        curr = head

        while curr:
            values.append(curr.val)
            curr = curr.next

        # 2. 双指针检查回文
        i, j = 0, len(values) - 1
        while i < j:
            if values[i] != values[j]:
                return False
            i += 1
            j -= 1
        return True
